clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLReader.java

```diff
@@ -81,10 +84,13 @@ public void readElement(XMLExtendedStreamReader reader, List<ModelNode> result)
             }
         }
 
+        // Explicitly order operations such that capabilities are defined before they are referenced
+        // This circumvents capability reference integrity issues when operations are not batched
```

Comment:
I thought about it later and my guess was it had something to do with subsystem tests. Do I win a prize?

This is definitely something we need to clean up as having this kind of thing proliferate through the codebase will be awful. But that's not this PR's problem.